Add support for setExceptionPauseMode

DanTup opened the pull request:
This adds a method to `Isolate` to set the exception mode and a test for each mode.

Fixes #37.

DanTup committed:
Add support for setExceptionMode

Fixes #37.

DanTup commented:
(I believe the Travis failure is because the tests are just broken in the latest SDKs. I ran against an older SDK and all is good. I think @kevmoo was setting this up recently and it's still WIP).

DanTup committed:
Fix formatting

jacob314 reviewed (approved):
Please wait for @nex3 to approve as well.

nex3 commented on lib/src/isolate.dart:
```diff
@@ -501,6 +506,26 @@ class VMStep {
+  /// The isolate will pause on all exceptions.
+  static const all = const VMExceptionPauseMode._("All");
+
+
```
Nit: extra newline

nex3 commented on lib/src/isolate.dart:
```diff
@@ -288,6 +288,11 @@ class VMIsolateRef {
+  Future setExceptionPauseMode(VMExceptionPauseMode mode) =>
+      _scope.sendRequest("setExceptionPauseMode", {"mode": mode._value});
+
+
```
Nit: extra newline

nex3 commented on lib/src/isolate.dart:
```diff
@@ -501,6 +506,26 @@ class VMStep {
+  static const all = const VMExceptionPauseMode._("All");
+
+
+  /// The string name of the step type.
```
"step type" is wrong here

nex3 reviewed (changes requested):
Since you're providing the ability to set this, you should probably add a getter as well.

nex3 commented:
One other thing: please make sure all the APIs you add document the version of the protocol in which they first appear.

DanTup committed:
Add ability to read exceptionPauseMode

+ review comments + changelog

DanTup commented:
I've added the ability to get the pause mode and an additional test. Also added to changelog and updated version (I'll merge this with the other PR if they're landed around the same time).

@nex3 The test for the getter currently calls `toString()` since the `VMExceptionPauseMode` instances are different. Let me know if you'd prefer this was done another way (I was thinking a factory constructor could use the string to return the same instances, or we could override equality - not really sure what the usual done thing is in Dart).

> One other thing: please make sure all the APIs you add document the version of the protocol in which they first appear.

As far as I can tell from the docs, exceptionPauseMode has always been there (it's not in the changelog), though the stuff in the other PR is new, so I'll go back and update that one.

DanTup commented:
(btw, someone will need to merge this when happy with it as I don't have merge permissions here)

DanTup commented:
I've rebased on master since the version number I used has been used in `master` now. Lmk if you'd like any further changes.
